### DISCUSSION TITLE
Tool or api to generate api document

### DISCUSSION THREAD

**Discussion by willser** [actix/actix-web#2723] [General] [answered]
Hi,this is a feature request and discussion.
I search this title in `discussion`, `issue` and `document`,but I got no result.Please forgive me if I miss them.

For a back-end developer,it's helpful if there are some `api` in `actix-web` or `external tool` to create `api document`.E.g,When a api be changed,the document of this api should be changed(In some workflows the correct approach is to change the document first).If there is a tool to generate api document,we can create api document in CI and push them to somewhere save or replace the order.

I thought about it,a way is parsing the code when building and print the infomation about `api`.Another way is add a new `api` like `/api/doc`,it be registered before `App` be created and running,it includ information about all api and could browse it .

Looking forward to your idea or discussion.

**Comment by robjtede** [comment, marked as answer]
There's the `paperclip` crate: https://github.com/paperclip-rs/paperclip

I'm also working on something similar that may help. I'll update if it's made public.

**Reply by willser** [reply]
Amazing.It's very helpful.Thanks for your contribution.
